### PR TITLE
Consistent naming for images kwargs

### PR DESCRIPTION
**Pull request by zucchini-nlp**
# What does this PR do?

it lays ground for my next PR, where I am aiming to unify `DefaultImagesKwargsFastInit`and `ImagesKwargs` since they are almost identical except for three kwargs (size_divisor, do_pad and pad_size)

This PR consolidates naming of the above there kwargs. Specifically:
- `size_divisor` is removed from general kwargs as it's used only in ~5 model
- `pad_size` and `do_pad` are added in default set of fast image processor kwargs along with a default `self.pad` method
- A few deprecations added where the naming was not consistent, e.g. `size_divisibility` instead of `size_divisor`

### CODE REVIEW

File: src/transformers/models/donut/image_processing_donut.py [hunk 1]
```diff
@@ -215,10 +215,6 @@ def pad_image(
         padding = ((pad_top, pad_bottom), (pad_left, pad_right))
         return pad(image, padding, data_format=data_format, input_data_format=input_data_format)
 
-    def pad(self, *args, **kwargs):
-        logger.info("pad is deprecated and will be removed in version 4.27. Please use pad_image instead.")
```
Comment (zucchini-nlp) [inline src/transformers/models/donut/image_processing_donut.py:219]: 4.27 🤯

**Comment by zucchini-nlp** [comment]
Done, finally! @yonigozlan , I'd like to ask a first review from you as it touches fast image processors. To keep you aligned, the next PR will remove `DefaultFastImageKwargs` and replace it with `ImagesKwargs` everywhere. I have a draft locally, haven't pushed changes to all models yet

File: src/transformers/image_processing_utils_fast.py [hunk 2]
```diff
@@ -181,6 +177,8 @@ class DefaultFastImageProcessorKwargs(TypedDict, total=False):
     do_normalize: Optional[bool]
     image_mean: Optional[Union[float, list[float]]]
     image_std: Optional[Union[float, list[float]]]
+    do_pad: Optional[bool]
+    pad_size: Optional[dict[str, int]]
```
Comment (yonigozlan) [inline src/transformers/image_processing_utils_fast.py:181]: I was a bit hesitant to do this because it seemed that the pad functions are very different across image processors (some pad to a fixed size, some pad to a divisibility, some pad only on the number of patches dimension etc.). Putting it as default also means using a default docstring for all these different cases, which might be misleading. 
Although I agree that your default pad function is the one that makes the most sense, and will probably simplify quite a few processors. Just wondering how we can ensure that we have correct docstring for the `do_pad` argument. Maybe make it vague in `auto_docstring` and invite the user to check the code to see what the padding function really does?
Happy to hear your thoughts on this!

File: src/transformers/image_processing_utils_fast.py [hunk 3]
```diff
@@ -239,6 +242,74 @@ def is_fast(self) -> bool:
...
+                stacked_masks = torch.zeros_like(stacked_images, dtype=torch.int64)[..., 0, :, :]
+                stacked_masks[..., : image_size[0], : image_size[1]] = 1
+                processed_masks_grouped[shape] = stacked_masks
+
+        processed_images = reorder_images(processed_images_grouped, grouped_images_index)
+        if return_mask:
+            processed_masks = reorder_images(processed_masks_grouped, grouped_images_index)
+            return processed_images, processed_masks
+
+        return processed_images
```
Comment (yonigozlan) [inline src/transformers/image_processing_utils_fast.py:311]: Nice!

File: src/transformers/models/deepseek_vl/image_processing_deepseek_vl_fast.py [hunk 4]
```diff
@@ -62,6 +62,7 @@ class DeepseekVLImageProcessorFast(BaseImageProcessorFast):
     do_resize = True
     do_rescale = True
     do_normalize = True
+    do_pad = True
```
Comment (yonigozlan) [inline src/transformers/models/deepseek_vl/image_processing_deepseek_vl_fast.py:65]: Did the processor never applied padding before? Weird if the tests were passing 😅

File: src/transformers/utils/auto_docstring.py [hunk 5]
```diff
@@ -131,6 +131,22 @@ class ImageProcessorArgs:
         "shape": None,
     }
 
+    do_pad = {
+        "description": """
+    Whether to pad the image to the images in the batch.
```
Comment (yonigozlan) [inline src/transformers/utils/auto_docstring.py]: Yes maybe here we can also guide the users to the code directly to see exactly what padding is done

**Review by yonigozlan — COMMENTED** [review]
Thanks a lot for the huge work!
First quick review, I didn't check every models, but overall I'm completely on board with this, very much needed.
Cant' wait to unify `DefaultFastImageProcessorKwargs` and `ImagesKwargs`

File: src/transformers/image_processing_utils_fast.py (hunk 3 above)
Comment (yonigozlan) [reply src/transformers/image_processing_utils_fast.py:311]: I was also wondering if we should include the whole group and reorder process in other functions like resize or rescale and normalize. Bu that means we can't have several processing functions under one grouping loop. And a new grouping is only needed if the previous function changed the shape of the images (not need after a rescale and normalize for example), and they do cost a bit of processing time, so it might be best to leave this to the `_preprocess` function. What do you think?

File: src/transformers/models/deepseek_vl/image_processing_deepseek_vl_fast.py (hunk 4 above)
Comment (zucchini-nlp) [reply src/transformers/models/deepseek_vl/image_processing_deepseek_vl_fast.py:65]: it was hardcoded and we always called padding irrespective of arguments haha

File: src/transformers/image_processing_utils_fast.py (hunk 2 above)
Comment (zucchini-nlp) [reply src/transformers/image_processing_utils_fast.py:181]: I had same thoughts because I wanted to delete all `self.pad` from models. I think there are two big patterns on padding:

1. The current way when we pad once and for all images at once, so they can be concatenated as tensors
2. The padding is applied as `pad_to_square` where each image is padded to be its own maximum size

So maybe we could still get a general method in base class if we allow each model to infer the pad size from smth like `self.get_pad_size(image, kwargs)`. I haven't looked through all models, but could be that pad size can be inferred similar to `self.size` when resizing. Then the most important thing will be to differentiate between calling pad either once per all images or per single image

Comment (zucchini-nlp) [reply src/transformers/image_processing_utils_fast.py:181]: For docs, I agree, we might have to keep it vague or override in models? Not sure if we can change docs if it's in auto

File: src/transformers/image_processing_utils_fast.py (hunk 3 above)
Comment (zucchini-nlp) [reply src/transformers/image_processing_utils_fast.py:311]: The grouping for padding was added only because it's supposed to be called outside of the loop if we want to pad all images to the same size. I see what you mean and it's partially related to above thread. We need to handle two types of paddings. WDYT of adding an arg like `group_images=False` ?

File: src/transformers/image_processing_utils_fast.py (hunk 2 above)
Comment (yonigozlan) [reply src/transformers/image_processing_utils_fast.py:181]: Yep we can override the doc, but I doubt contributors will do it and reviewers will catch it consistently tbh

File: src/transformers/image_processing_utils_fast.py (hunk 3 above)
Comment (yonigozlan) [reply src/transformers/image_processing_utils_fast.py:311]: Oh yes that makes sense then, if we're padding to the maximum size. I don't think we need to add anything, we can just override the function when we have a different type of padding needed

File: src/transformers/models/deepseek_vl/image_processing_deepseek_vl_fast.py (hunk 4 above)
Comment (yonigozlan) [reply src/transformers/models/deepseek_vl/image_processing_deepseek_vl_fast.py:65]: Oh I see haha

File: src/transformers/image_processing_utils_fast.py (hunk 3 above)
Comment (zucchini-nlp) [reply src/transformers/image_processing_utils_fast.py:311]: yep, that is what most processors do. Though it's not "overriding", they use fn called `pad_to_square` which we can actually standardize and add in Base class as well in the next iteration

**Review by Cyrilvallez — APPROVED** [review]
Very nice! Aligned with the standardization vision we have for v5! I'm just confused about the default value of `None`. Why not just set it to False so it's a proper bool, instead of relying on the truth value of None being False?

**Comment by zucchini-nlp** [comment]
Will merge to unblock myself for the next PR

EDIT: I don''t know where my comment is gone with a reply. Anyway, the `None` is simply for consistency with how fast image processors are implemented and other kwargs. We can batch update the defaults in subsequent PRs
